vcd/resource_vcd_vapp_vm.go

```diff
@@ -143,6 +144,22 @@ var vappVmSchema = map[string]*schema.Schema{
 		Computed:    true,
 		Description: "Storage profile to override the default one",
 	},
+	"os_type": &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Operating System type. Possible values can be found in documentation.",
+	},
+	"hardware_version": &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    true,
+		Computed:    true,
+		Description: "Virtual Hardware Version (e.g.`vmx-14`, `vmx-13`, `vmx-12`, etc.)",
+	},
+	"boot_image": &schema.Schema{
```

Comment:
I also wondered regarding, but decided to leave as optional and allow to change to empty. In this way I unmount it. So user has ability, otherwise he could do that only manually in UI or other means.

Reply:
Ok, that makes sense actually. One will probably want to detach image after installation.